### PR TITLE
Weekly digest email

### PR DESCRIPTION
**Pull request by RubenSousaDinis**
## Summary

This PR adds the new digest email.

## Notion link

https://www.notion.so/talentprotocol/Platform-Emails-9bd47f988fb947d7aa0e775dfa036499

### CODE REVIEW

File: app/views/user_mailer/send_digest_email.html.erb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    </p>
+    <% @talents.each do |talent| %>
+      <%= render "shared_mailer/talent", talent: talent %>
+    <% end %>
+    <p class="w-full lh-lg text-lg mb-10 primary-3 paragraph">
+      have just launched their tokens! Check out their profiles and let's make collective success a reality.
+    </p>
+    <%= link_to "Find new Talent", talent_index_url(status: "Latest Added"), class: "btn btn-primary fw-800 text-5xl py-4 mb-5 w-full" %>
+  </div>
+<% end %>
```
Comment (verdefred) [inline app/views/user_mailer/send_digest_email.html.erb:68]: end of file missing newline

File: db/schema.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@
 ActiveRecord::Schema.define(version: 2022_07_13_152311) do
 
   # These are extensions that must be enabled in order to support this database
+  enable_extension "pg_stat_statements"
```
Comment (verdefred) [inline db/schema.rb:16]: is this from this PR?

**Review by verdefred — COMMENTED** [review]
Great work just left a couple of comments

File: app/jobs/email_reminders/send_digest_email_job.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  class SendDigestEmailJob < ApplicationJob
+    queue_as :default
+
+    def perform
+      users = User.joins(:talent)
+        .where("users.digest_email_sent_at < ? OR users.digest_email_sent_at IS NULL", ENV["DIGEST_EMAIL_DAYS"].to_i.days.ago)
+
+      unless ENV["DIGEST_EMAIL_ENABLED"] == "true"
+        users = users.where(role: "admin")
+      end
```
[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply app/jobs/email_reminders/send_digest_email_job.rb:11]: This is only a filter for now. When we enable the email all users will receive it. For now only admins will get it

File: app/assets/stylesheets/bootstrap-email.scss [hunk 4]
```diff
@@ -104,6 +104,14 @@ $light-grey: #a9acb7 !default;
   padding-right: 0 !important;
 }
 
+.ml-1 {
+  margin-left: 8px !important;
```
[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply app/assets/stylesheets/bootstrap-email.scss:108]: makes sense. Will change

File: db/schema.rb (hunk 2 above)
Comment (RubenSousaDinis) [reply db/schema.rb:16]: will remove

File: app/mailers/user_mailer.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 class UserMailer < ApplicationMailer
+  TAL_DECIMALS = 10**18
+  TAL_VALUE_IN_USD = 0.02
```
Comment (francisco-leal) [inline app/mailers/user_mailer.rb:3]: Can we put this consts in the Token model? I think they make more sense to exist there

File: app/mailers/user_mailer.rb [hunk 6]
```diff
@@ -54,9 +57,46 @@ def send_complete_profile_reminder_email
     bootstrap_mail(to: user.email, subject: "Complete your profile and launch your token today 🚀")
   end
 
+  def send_digest_email
+    @user = params[:user]
+    @without_container = true
```
Comment (francisco-leal) [inline app/mailers/user_mailer.rb:62]: When should this be false? I don't see it referenced in other functions

File: app/jobs/email_reminders/send_digest_email_job.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+      users.each do |user|
+        UserMailer.with(user: user).send_digest_email.deliver_later
+        user.update!(digest_email_sent_at: Time.zone.now)
```
Comment (francisco-leal) [inline app/jobs/email_reminders/send_digest_email_job.rb:15]: Should this update happen within the user mailer logic?

File: app/views/shared_mailer/_talent.html.erb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<% profile_picture_url = attachments["profile_picture-#{talent.id}.png"]&.url %>
+<% user = talent.user %>
```
Comment (francisco-leal) [inline app/views/shared_mailer/_talent.html.erb:2]: dont we have a `@user` variable in this scope already?

File: app/views/user_mailer/send_digest_email.html.erb (hunk 1 above)
Comment (francisco-leal) [inline app/views/user_mailer/send_digest_email.html.erb:68]: missing new line

File: app/jobs/email_reminders/send_digest_email_job.rb (hunk 7 above)
Comment (RubenSousaDinis) [reply app/jobs/email_reminders/send_digest_email_job.rb:15]: I like to treat mailer functions like controller methods. Business logic should not be there. But I see your point. The email can fail and the update is still performed. If the email fail we will still be able to resend it from Sidekiq.

File: app/views/shared_mailer/_talent.html.erb (hunk 8 above)
Comment (RubenSousaDinis) [reply app/views/shared_mailer/_talent.html.erb:2]: We have, but it's a good practice not to use instance variables in partials. That's why I have passed variables to the partial and I only use them.

Comment (RubenSousaDinis) [reply app/views/shared_mailer/_talent.html.erb:2]: Other mailer with a @talent variable only will not be able to reuse it then.

File: app/mailers/user_mailer.rb (hunk 6 above)
Comment (RubenSousaDinis) [reply app/mailers/user_mailer.rb:62]: False is the default state. This is the only exception. Only this emails have a fluid container.

File: app/mailers/user_mailer.rb (hunk 5 above)
Comment (RubenSousaDinis) [reply app/mailers/user_mailer.rb:3]: Agree!

File: db/schema.rb (hunk 2 above)
Comment (RubenSousaDinis) [reply db/schema.rb]: done

File: app/mailers/user_mailer.rb (hunk 5 above)
Comment (RubenSousaDinis) [reply app/mailers/user_mailer.rb]: Done

File: app/assets/stylesheets/bootstrap-email.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply app/assets/stylesheets/bootstrap-email.scss]: changed

**Review by verdefred — APPROVED** [review]

**Pull request merged**